google-map-poly.html

```diff
@@ -0,0 +1,554 @@
+      this.poly && this.poly.set('strokePosition', this._convertStrokePosition());
+    },
+
+    _strokeWeightChanged: function() {
+      this.poly && this.poly.set('strokeWeight', this.strokeWeight);
+    },
+
+    _zIndexChanged: function() {
+      this.poly && this.poly.set('zIndex', this.zIndex);
+    },
+
+    // Helper logic.
+
+    _buildPathFromPoints: function() {
+      this._points = Array.prototype.slice.call(Polymer.dom(this.$.points).getDistributedNodes());
+      window.console.log("Points:  " + this._points.length);
+      window.console.log(Polymer.dom(this).getDestinationInsertionPoints());
+
+      // Build path from current points (ignoring vertex insertions while doing so).
+      this._building = true;
+      this.path.clear();
+      for (var i = 0, point; point = this._points[i]; ++i) {
+        this.path.push(point.getPosition());
+      }
+      this._building = false;
+
+      this.fire('google-map-poly-path-built', this.path);
+
+      // Watch for future updates.
+      if (this._mutationObserver) {
+        return;
+      }
+      this._mutationObserver = new MutationObserver(this._buildPathFromPoints.bind(this));
+      this._mutationObserver.observe(this, {
+        childList: true
+      });
+    },
+
+    _clearListener: function(name) {
+      if (this._listeners[name]) {
+        google.maps.event.removeListener(this._listeners[name]);
+        this._listeners[name] = null;
+      }
+    },
+
+    _convertStrokePosition: function() {
+      return google.maps.StrokePosition && this.strokePosition?
+          google.maps.StrokePosition[this.strokePosition.toUpperCase()] : 0;
+    },
+
+    _createPoly: function() {
+      // Build poly's path and register mutation listeners on first creation.
+      if (!this.path) {
+        this._setPath(new google.maps.MVCArray());
+        google.maps.event.addListener(this.path, 'insert_at', this._startEditing.bind(this));
+        google.maps.event.addListener(this.path, 'set_at', this._updatePoint.bind(this));
+        this._buildPathFromPoints();
+      }
+
+      var options = {
+        clickable: this.clickable || this.draggable,  // draggable must be clickable to work.
+        draggable: this.draggable,
+        editable: this.editable,
+        geodesic: this.geodesic,
+        map: this.map,
+        path: this.path,
+        strokeColor: this.strokeColor,
+        strokeOpacity: this.strokeOpacity,
+        strokePosition: this._convertStrokePosition(),
+        strokeWeight: this.strokeWeight,
+        visible: !this.hidden,
+        zIndex: this.zIndex
+      };
+
+      if (this.closed) {
+        options.fillColor = this.fillColor;
+        options.fillOpacity = this.fillOpacity;
+        this._setPoly(new google.maps.Polygon(options));
+      } else {
+        options.icons = this.icons;
+        this._setPoly(new google.maps.Polyline(options));
+      }
+
+      this._listeners = {};
+      this._clickEventsChanged();
```

Comment:
Check that these don't double call on page load. e.g. someone uses the `click-events` attribute. Won't this call and the natural `_clickEventsChanged()` setup double listeners?